ui/app/scripts/controllers/serviceVersionStatusCtrl.js

```diff
@@ -8,157 +8,124 @@ angular.module('apollo')
-            $scope.websocket = null;
-            $scope.term = null;
-            $scope.websocketScope = null;
-            $scope.terminalHeader = null;
-
-            $scope.showByService = function(service) {
-                usSpinnerService.spin('result-spinner');
-                $scope.filteredResults = [];
-                apolloApiService.serviceStatus(service.id).then(function (response) {
-                    $scope.filteredResults = response.data;
-                    $scope.filteredResults = $scope.filteredResults.filter(function(x){ return x !== null; });
-                    populateDeployableVersions();
-                    refreshPreSelectedStatus();
-
-                    if (service.isPartOfGroup) {
-                        var groupedByEnvironment = [];
-                        $scope.filteredResults.forEach(function (kubeDeploymentStatus) {
-                            var first = true;
-                            groupedByEnvironment.forEach(function (existingKubeDeploymentStatus) {
-                                if (existingKubeDeploymentStatus.environmentId === kubeDeploymentStatus.environmentId) {
-                                    first = false;
-                                    existingKubeDeploymentStatus.nestedKubeDeploymentStatuses.push(kubeDeploymentStatus);
-                                }
-                            });
-                            if (first) {
-                                kubeDeploymentStatus.nestedKubeDeploymentStatuses = [kubeDeploymentStatus];
-                                groupedByEnvironment.push(kubeDeploymentStatus);
-                            }
-                        });
-                        $scope.filteredResults = groupedByEnvironment;
-                    } else {
-                        $scope.filteredResults.forEach(function (kubeDeploymentStatus) {
-                            kubeDeploymentStatus.nestedKubeDeploymentStatuses = [];
+
+            $scope.setSelectedGroup = function(group) {
+                $scope.selectedGroup = group;
+            }
+
+            $scope.showAllGroupsOfService = function() {
+                if ($scope.selectedService != null && $scope.selectedEnvironment != null) { //selected service and environment
+                    if ($scope.selectedService.isPartOfGroup) { //service is part of group
+                        apolloApiService.allGroupsStatusesOfEnvironmentAndService($scope.selectedEnvironment.id, $scope.selectedService.id).then(function (response) {
+                            $scope.kubernetesDeploymentStatus = response.data;
+                            $scope.currentScreen = "results";
+                            }, function (error) {
+                                growl.error("An error occurred in getting k8s status", {ttl:7000});
                         });
+                    } else if ($scope.selectedEnvironment != null) { //error in case of unselected environment
```

Comment:
This is inside an "if" with the same condition, it can't happen

Reply:
Doesn't suppose to happen, unless something went wrong